retrieval/scrape.go

```diff
@@ -496,10 +496,10 @@ func (sl *scrapeLoop) report(start time.Time, duration time.Duration, err error)
 		Value:     model.SampleValue(duration.Seconds()),
 	}
 
-	if err := sl.reportAppender.Append(healthSample); err != nil {
+	if err := sl.reportAppender.Append(context.TODO(), healthSample); err != nil {
 		log.With("sample", healthSample).With("error", err).Warn("Scrape health sample discarded")
 	}
-	if err := sl.reportAppender.Append(durationSample); err != nil {
+	if err := sl.reportAppender.Append(context.TODO(), durationSample); err != nil {
```

Comment:
True, we can't use cancellations or timeouts here. I could change it from `context.TODO()` to just `context.Background()`, but the reason for the `TODO` was that we might at some point want to pass other metadata from up the callchain into the appender via a context. Either in Prometheus itself, or when someone reuses the retrieval package and maybe runs multiple scrape pools for different users, but only one appender. The `TODO` is stating "The context begins here for now, but it make sense to connect it up with a context coming from an upstream caller later".

I'm unsure what's best. FWIW, we don't need this in Frankenstein on the retrieval side because every user runs their own binary with the right configuration anyways (and the user config only happens in the appender anyways).

Options:

a) Don't contextify `Append()` for now. Would be a bit inconsistent with other storage methods (but at least the `SampleAppender` is a separate interface and just included into the `Storage` one).
b) Keep the `TODO()`s to signal that we want to connect it to upstream later.
c) Change to `Background()` to make sure nobody tries cancellations/timeouts on it.

For c), even if someone used a timeout or cancel on that context, the code here would simply have to ignore it (or if `Append()` ever got changed to take multiple samples, it would have to also either ignore the cancellation or support rolling everything back atomically).

Still leaning towards b), but fine with the other options as well. Or maybe there's a d) that eludes me...